Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -65,9 +73,9 @@ void GridMasonryLayout::collectMasonryItems()
         auto gridArea = grid.gridItemArea(*child);
         if (m_firstTrackItems.size() != m_gridAxisTracksCount && itemGridAreaStartsAtFirstLine(gridArea, m_masonryAxisDirection) && m_renderGrid.itemGridAreaIsWithinImplicitGrid(gridArea, m_gridAxisTracksCount + 1, gridAxisDirection()))
             m_firstTrackItems.add(child, gridArea);
-        else if (hasDefiniteGridAxisPosition(child, gridAxisDirection()))
+        else if (m_renderGrid.style().masonryAutoFlow().contains(MasonryAutoFlow::DefiniteFirst) && hasDefiniteGridAxisPosition(child, gridAxisDirection()))
```

Comment:
To avoid repeating the condition, this could be

```cpp
else if (m_renderGrid.style().masonryAutoFlow().contains(MasonryAutoFlow::DefiniteFirst)) {
    if (hasDefiniteGridAxisPosition(child, gridAxisDirection()))
        m_itemsWithDefiniteGridAxisPosition.append(child);
    else
        m_itemsWithIndefiniteGridAxisPosition.append(child);
}
```

Reply:
Done